### PR TITLE
feat: Backlog/unreal/project context/implement snapshot assets

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves 

- [ ] I have added automatic tests where applicable.
- [ ] The PR contains a description of what has been changed.
- [ ] The description contains manual test instructions.
- [X] The PR contains updates to the release notes.
- [ ] I have verified that the documentation is still up to date.

## Changes

- Publish Dependencies Asset info as metadata of the current asset [Batch publisher]
- Create asset publisher [Publisher]
- Implement asset tabs to AM [Unreal Assets] [pipeline Assets]
- Create snapshot asset info on dependency batch publish
- Check modified dependencies [Publisher]
- Recursively import dependencies [Loader, Opener]
- Check asset loader (uasset) [Loader]

## Test

_Note: has to be tested with corresponding branch on pipeline, definition, qt and unreal_

- Load FBX > set material > publish asset
- Open asset in new project, material should be imported with asset.
- Open level task > load asset in assembler > check AM that all assets are listed.

### CODE REVIEW

File: source/ftrack_connect_pipeline/asset/__init__.py [hunk 1]
```diff
@@ -202,3 +207,84 @@ def create_new_dcc_object(self):
...
+        result[asset_const.COMPONENT_ID] = component_id
+        result[asset_const.COMPONENT_NAME] = component_name
+        mod_date = None
+        file_size = None
+        if os.path.exists(local_component_path):
+            mod_date = os.path.getmtime(local_component_path)
+            file_size = os.path.getsize(local_component_path)
+        result[asset_const.MOD_DATE] = mod_date
+        result[asset_const.FILE_SIZE] = file_size
+        return result
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline/asset/__init__.py:290]: I think we should remove this from here and do the following:

- Create a "unreal_dependencytrack_publisher_post_finalizer_options" widget where by default we set the following options:
{     
     Plugin_name: unreal_native_loader_importer
     plugin_type: loader.importer (Not sure, maybe we can remove this option and set it automatically)
     method: init_and_load
     definition: asset Loader
}

- Then refactor the asset_info.py "generate_asset_info_dict_from_args" to accept passing the component_path (or create a new method)

File: source/ftrack_connect_pipeline/plugin/load/importer.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def load_asset(self, context_data=None, data=None, options=None):
         result = {
             'asset_info': self.asset_info,
             'dcc_object': self.dcc_object,
-            'run_method': run_result,
+            'run_result': run_result,
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline/plugin/load/importer.py:121]: Did you check what the run_result returned? 
Basically check if the result is just the plugin result or the result is something like:
{
run: "result of the run method",
fetch: "result of the fetch method",
}
If the result is this one, then run method is correct.

**Review by lluisFtrack — CHANGES_REQUESTED** [review]

File: source/ftrack_connect_pipeline/plugin/load/importer.py (hunk 2 above)
Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline/plugin/load/importer.py:121]: Yes I checked this and run_result contained the raw output from definition importer plugin, eg. {"<component path>":"<imported path>"}

Comment (lluisFtrack) [reply source/ftrack_connect_pipeline/plugin/load/importer.py:121]: So it contains the method? {"<method_name>":"<result>"} Then is better to name it run_method_result.

**Review by lluisFtrack — APPROVED** [review]
LGTM

**Pull request merged**